angular-workspace/projects/ni/nimble-angular/src/directives/anchor-button/nimble-anchor-button-router-link-with-href.directive.ts

```diff
@@ -20,4 +22,16 @@ export class NimbleAnchorButtonRouterLinkWithHrefDirective extends RouterLinkWit
     public set nimbleRouterLink(commands: never[] | string | null | undefined) {
         this.routerLink = commands;
     }
+
+    public constructor(injector: Injector, private readonly elementRef: ElementRef<AnchorButton>) {
```

Comment:
We have a directive base for anchors generally: https://github.com/ni/nimble/blob/main/angular-workspace/projects/ni/nimble-angular/src/directives/anchor-base/nimble-anchor-base.directive.ts

Seems like we could have a `DisableableRouterLinkWithHref extends RouterLinkWithHref`  base all these could extend from?

You'll probably have to make the ElementRef type generic, something like `T extends {disabled: boolean}` and can then do something like `NimbleAnchorButtonRouterLinkWithHrefDirective extends DisableableRouterLinkWithHref<AnchorButton>`, etc.